actors/miner/tests/util.rs

```diff
@@ -1110,17 +1112,185 @@ impl ActorHarness {
+            pieces
+                .iter()
+                .filter_map(|p| {
+                    p.verified_allocation_key.as_ref().map(|a| AllocationClaim {
+                        client: a.client,
+                        allocation_id: a.id,
+                        data: p.cid,
+                        size: p.size,
+                    })
+                })
+                .collect()
+        }
+        fn notifications_from_pieces(pieces: &[PieceActivationManifest]) -> Vec<PieceChange> {
+            pieces
+                .iter()
+                .map(|p| {
+                    p.notify.iter().map(|n| PieceChange {
+                        data: p.cid,
+                        size: p.size,
+                        payload: n.payload.clone(),
+                    })
+                })
+                .flatten()
+                .collect()
+        }
+
+        let params = ProveReplicaUpdates3Params {
+            sector_updates: sector_updates.into(),
+            sector_proofs: sector_updates.iter().map(|su| make_proof(su.sector as u8)).collect(),
+            aggregate_proof: RawBytes::default(),
+            update_proofs_type: self.seal_proof_type.registered_update_proof().unwrap(),
+            aggregate_proof_type: RegisteredAggregateProof::Invalid(0),
+            require_activation_success,
+            require_notification_success,
+        };
+        rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, self.worker);
+        rt.expect_validate_caller_addr(self.caller_addrs());
+
+        let mut sector_allocation_claims = Vec::new();
+        let mut sector_claimed_space = Vec::new();
+        let mut expected_pledge = TokenAmount::zero();
+        let mut expected_qa_power = StoragePower::zero();
+        let mut expected_sector_notifications = Vec::new(); // Assuming all to f05
+        for sup in sector_updates {
+            let sector = self.get_sector(rt, sup.sector);
+            let expected_unsealed_cid_inputs: Vec<PieceInfo> =
+                sup.pieces.iter().map(|p| PieceInfo { size: p.size, cid: p.cid }).collect();
+            let unsealed_cid = make_piece_cid(format!("unsealed-{}", sup.sector).as_bytes());
+            rt.expect_compute_unsealed_sector_cid(
+                self.seal_proof_type,
+                expected_unsealed_cid_inputs,
+                unsealed_cid.clone(),
+                ExitCode::OK,
+            );
+            rt.expect_replica_verify(
+                ReplicaUpdateInfo {
+                    update_proof_type: self.seal_proof_type.registered_update_proof().unwrap(),
+                    new_sealed_cid: sup.new_sealed_cid,
+                    old_sealed_cid: sector.sealed_cid,
+                    new_unsealed_cid: unsealed_cid,
+                    proof: make_proof(sup.sector as u8).into(),
+                },
+                Ok(()),
+            );
+            let claims = SectorAllocationClaims {
```

Comment:
We'll want to share large parts of this with the prove commit 2 harness method.  I'll look into creating shared functions in a parallel PR that you can take or leave for this one.

Reply:
I've extracted some of the inner helper functions to top level.